core/src/test/java/org/apache/iceberg/TestMetrics.java

```diff
@@ -72,23 +78,14 @@
 
   public abstract File writeRecords(Schema schema, Record... records) throws IOException;
 
+  public abstract File writeRecords(Schema schema, Map<String, String> properties, GenericData.Record... records)
+      throws IOException;
+
+  public abstract int getRowGroupSize(File parquetFile) throws IOException;
```

Comment:
I think this is actually row group count, right? In that case, I think this should be named with "count" instead of "size".

Also, this is a generic metrics test so this should not be specific to Parquet. How about `splitCount(File dataFile)` instead?

Reply:
refactored.